lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3808,30 +3808,30 @@ class ConvertAtenSizeIntOp : public OpConversionPattern<AtenSizeIntOp> {
 };
 } // namespace
 
-// Casts a 0d integer tensor to elemental type.
 namespace {
-class ConvertAtenIntTensorOp : public OpConversionPattern<AtenIntTensorOp> {
+// Casts a 0d tensor to elemental type.
+template <typename OpTy>
+class ConvertAtenTensorToScalarLikeOp : public OpConversionPattern<OpTy> {
 public:
-  using OpConversionPattern::OpConversionPattern;
+  using OpConversionPattern<OpTy>::OpConversionPattern;
   LogicalResult
-  matchAndRewrite(AtenIntTensorOp op, OpAdaptor adaptor,
+  matchAndRewrite(OpTy op, typename OpTy::Adaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
     if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
       return failure();
-    Value intTensor = adaptor.a();
-    auto tensorType = intTensor.getType().cast<RankedTensorType>();
+    Value input = adaptor.a();
+    auto inputType = input.getType().cast<RankedTensorType>();
 
-    if (tensorType.getRank() != 0)
-      return rewriter.notifyMatchFailure(
-          op, "invalid rank: the rank of the input tensor must be 0");
+    if (inputType.getRank() != 0)
```

Comment:
The rank can be more than rank 0. It's ok as long as there is only one element.

Reply:
Thanks for this point :)